### PR TITLE
docker: add docker socket help for mac users

### PR DESCRIPTION
**Pull request by tmorrell**
:heart: Thank you for your contribution!

### Description

Adds some help info for diagnosing problems with Docker on a Mac.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've targeted the `master` branch.
- [x] If this documentation change impacts the current release of InvenioRDM, I will backport it to the `production` branch following approval or indicate to a maintainer that it should be backported.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: docs/install/requirements.md [hunk 1]
```diff
@@ -95,6 +95,15 @@ and adjust the `Memory` to the corresponding value. If you have a few cores
 more to spare, it might be a good idea to give more than 2. Take into account
 that you will run between 4 and 8 containers.
 
+#### Docker socket (macOS)
+
+invenio-cli uses the Docker Python API to check things like the Docker version. The Docker Python API
+uses the Docker socket, which may not be enabled by default on a Mac. You'll know you have this problem
```
Comment (fenekku) [inline docs/install/requirements.md:101]:
```suggestion
uses the Docker socket, which may not be enabled by default in Docker Desktop on a Mac. You'll know you have this problem
```

:heart: Yes, colleague ran into same issue last week.

File: docs/install/requirements.md [hunk 2]
```diff
@@ -95,6 +95,15 @@ and adjust the `Memory` to the corresponding value. If you have a few cores
...
+if you see the error message `docker.errors.DockerException: Error while fetching server API version:`.
+You can enable to Docker socket by going to Docker desktop, Settings, Advanced, and checking the box for
```
Comment (fenekku) [inline docs/install/requirements.md:103]:
```suggestion
You can enable the Docker socket by going to Docker Desktop > Settings > Advanced, and checking the box for 
```

**Review by fenekku — APPROVED** [review]
Thanks! Pinging @meowcenary who ran into this issue as well. Is that what worked for you Eric?

**Comment by Meowcenary** [comment]
> Thanks! Pinging @Meowcenary who ran into this issue as well. Is that what worked for you Eric?

This was definitely an issue I ran into at some point in the past, but most recently it was just user error. I can't remember exactly what the underlying issue was, but I had Postgres running as a Homebrew service and that was interfering with the Postgres container.

**Comment by fenekku** [comment]
Can I let you backport this @tmorrell ? Thanks!

**Pull request merged**